src/test/test-models/good/function-signatures/math/matrix/add_diag.stan

```diff
@@ -0,0 +1,18 @@
+functions {}
+data {
+  int<lower=1> N;
+  matrix[N, N] K;
+  row_vector[N] row_vec;
+  vector[N] vec;
+  int scalar;
+}
+transformed data {
```

Comment:
This file should have tests for all possible combinations of parameter and data for all possible signatures.   That's a total of 3 signatures * 4 instantiations each (data/data, data/parameter, parameter/data, parameter/parameter).